src/Symfony/Bridge/Twig/AppVariable.php

```diff
@@ -174,6 +174,13 @@ public function getFlashes($types = null)
             return $session->getFlashBag()->get($types);
         }
 
-        return array_intersect_key($session->getFlashBag()->all(), array_flip($types));
+        $result = array();
+        foreach ($types as $type) {
+            if ($value = $session->getFlashBag()->get($type)) {
```

Comment:
I did not notice this behaviour, but according to the goal of the initial PR to make the code in Twig templates easier to use I think we should make that change.